test/Remittance.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+const Remittance = artifacts.require('Remittance');
+const truffleAssert = require('truffle-assertions');
+
+const {toWei, fromWei } = web3.utils;
+const BN = require('big-number');
+
+contract('Remittance', (accounts) => {
+    const owner = accounts[0];
+    const exchange = accounts[1];
+    const stranger = accounts[2];
+    const nil = "0x0000000000000000000000000000000000000000";
+    const password = web3.utils.utf8ToHex("123456");
+    let instance;
+
+    const evmMethod = (method, params = []) => {
+        return new Promise(function (resolve, reject) {
+            const sendMethod = (web3.currentProvider.sendAsync) ? web3.currentProvider.sendAsync.bind(web3.currentProvider) : web3.currentProvider.send.bind(web3.currentProvider);
+            sendMethod(
+                {
+                    jsonrpc: '2.0',
+                    method,
+                    params,
+                    id: new Date().getSeconds()
+                },
+                (error, res) => {
+                    if (error) {
+                        return reject(error);
+                    }
+    
+                    resolve(res.result);
+                }
+            );
+        });
+    };
+    
+    const increaseTime = async (amount) => {
+        await evmMethod("evm_increaseTime", [Number(amount)]);
+        await evmMethod("evm_mine");
+    };
+
+    beforeEach("Creating new contract", async () => {
+        instance = await Remittance.new({from: owner});
+    });
+
+    describe('======= newTransaction unit testing =======', () => {
+        it('Should create a new transaction with no extra fees', async () => {
+            const txObj = await instance.newTransaction(exchange, password, {from: owner, value: toWei("0.1", "ether")});
+            assert.strictEqual(txObj.logs.length, 1);
+            assert.strictEqual(txObj.logs[0].event, "LogNewTransaction");
+            assert.strictEqual(txObj.logs[0].args[0], owner);
+            assert.strictEqual(txObj.logs[0].args[1], exchange);
+            assert.strictEqual(fromWei(txObj.logs[0].args[2], "ether"), "0.1");
+    
```

Comment:
Also test that storage is updated as expected.

Reply:
How can I check the storage considering that the map is private?

Reply:
If you want to make it easy, make it `public`.
If you want to make it difficult, look at how the location is calculated with `sha3`, and then use `web3.eth.getStorageAt`.